packages/flutter/lib/src/material/text_form_field.dart

```diff
@@ -72,4 +74,69 @@ class TextFormField extends FormField<String> {
       );
     },
   );
+
+  /// Controls the text being edited.
+  ///
+  /// If null, this widget will create its own [TextEditingController].
+  final TextEditingController controller;
+
+  @override
+  _TextFormFieldState createState() => new _TextFormFieldState();
+}
+
+class _TextFormFieldState extends FormFieldState<String> {
+  TextEditingController _controller;
+
+  TextEditingController get _effectiveController => widget.controller ?? _controller;
+
+  @override
+  TextFormField get widget => super.widget;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.controller == null) {
+      _controller = new TextEditingController(text: widget.initialValue);
+    } else {
+      widget.controller.text = widget.initialValue;
```

Comment:
Add a comment here justifying why you can write to this object that you don't own
Also, please make sure TextEditingController documents whether it's even remotely ok for one TextEditingController to be handed to multiple text fields at once. If it's not, please make sure we assert somehow that that is not happening.

Reply:
Was discussing the initialValue/controller consistency issue the other day (for unrelated reasons) and I wonder whether a better alternative might be that we define `initialValue` to be the the initial value of the field, but that passing in a controller is equivalent to instantly editing the value to something other than the default value.

That seems relatively intuitive, but would obviously need to be clearly documented. Thoughts?

Reply:
Updated the class and constructor doc comments to explicitly document this behaviour.